packages/protocol/engine/tangle/booker/markerbooker/booker.go

```diff
@@ -364,9 +357,25 @@ func (b *Booker) GetAllAttachments(txID utxo.TransactionID) (attachments *advanc
 	return b.attachments.GetAttachmentBlocks(txID)
 }
 
+// storeNewBlock tries to store a new Block in the in-memory storage of the Booker and returns if the Block was stored.
+func (b *Booker) storeNewBlock(block *booker.Block) bool {
+	b.evictionMutex.Lock()
+	defer b.evictionMutex.Unlock()
```

Comment:
This used to be a RLock

```suggestion
	b.evictionMutex.RLock()
	defer b.evictionMutex.RUnlock()
```